src/main/java/me/ryanhamshire/GriefPrevention/PlayerEventHandler.java

```diff
@@ -1705,47 +1706,15 @@ void onPlayerInteract(PlayerInteractEvent event)
-                                clickedBlockType == Material.DARK_OAK_DOOR)) ||
-
-                (instance.config_claims_preventButtonsSwitches && (clickedBlockType == Material.WHITE_BED ||
-                        clickedBlockType == Material.ORANGE_BED ||
-                        clickedBlockType == Material.MAGENTA_BED ||
-                        clickedBlockType == Material.LIGHT_BLUE_BED ||
-                        clickedBlockType == Material.YELLOW_BED ||
-                        clickedBlockType == Material.LIME_BED ||
-                        clickedBlockType == Material.PINK_BED ||
-                        clickedBlockType == Material.GRAY_BED ||
-                        clickedBlockType == Material.LIGHT_GRAY_BED ||
-                        clickedBlockType == Material.CYAN_BED ||
-                        clickedBlockType == Material.PURPLE_BED ||
-                        clickedBlockType == Material.BLUE_BED ||
-                        clickedBlockType == Material.BROWN_BED ||
-                        clickedBlockType == Material.GREEN_BED ||
-                        clickedBlockType == Material.RED_BED ||
-                        clickedBlockType == Material.BLACK_BED)) ||
-
-                (instance.config_claims_lockTrapDoors && (
-                        clickedBlockType == Material.OAK_TRAPDOOR ||
-                                clickedBlockType == Material.SPRUCE_TRAPDOOR ||
-                                clickedBlockType == Material.BIRCH_TRAPDOOR ||
-                                clickedBlockType == Material.JUNGLE_TRAPDOOR ||
-                                clickedBlockType == Material.ACACIA_TRAPDOOR ||
-                                clickedBlockType == Material.DARK_OAK_TRAPDOOR)) ||
-
-                (instance.config_claims_lockFenceGates && (
-                        clickedBlockType == Material.OAK_FENCE_GATE ||
-                                clickedBlockType == Material.ACACIA_FENCE_GATE ||
-                                clickedBlockType == Material.BIRCH_FENCE_GATE ||
-                                clickedBlockType == Material.JUNGLE_FENCE_GATE ||
-                                clickedBlockType == Material.SPRUCE_FENCE_GATE ||
-                                clickedBlockType == Material.DARK_OAK_FENCE_GATE)) ||
-                (instance.config_claims_lecternReadingRequiresAccessTrust && clickedBlockType == Material.LECTERN))
+                (instance.config_claims_lockWoodenDoors && Tag.WOODEN_DOORS.isTagged(clickedBlockType) ||
+
+                instance.config_claims_preventButtonsSwitches && Tag.BEDS.isTagged(clickedBlockType) ||
+
+                instance.config_claims_lockTrapDoors && Tag.WOODEN_TRAPDOORS.isTagged(clickedBlockType) ||
+
+                instance.config_claims_lecternReadingRequiresAccessTrust && clickedBlockType == Material.LECTERN ||
+
+                instance.config_claims_lockFenceGates && clickedBlock.getBlockData() instanceof Gate))
```

Comment:
Forgot to mention this - I checked Gate because the setting is specifically for that, but it may be worth considering whether or not people actually ever separate wooden door, gate, and trapdoor settings. May be able to get away with both simplifying configuration and minimizing checks by just having one setting for Openable.

If Mojang ever adds a Gate tag out of the box, the current approach is more efficient, but we could still simplify the config checks.

Reply:
One place to check would be the stats, but at least for me you're right. I did have to make an addon for trapdoors over ice though, since because they're cheaper to produce, most players at least on my server use wooden trapdoors for ice corridors.